mountpoint-s3/src/fs.rs

```diff
@@ -130,14 +130,14 @@ where
         pid: u32,
         fs: &S3Filesystem<Client, Prefetcher>,
     ) -> Result<FileHandleType<Client, Prefetcher>, Error> {
-        // We don't need to do bit masking here because O_RDONLY value is zero.
-        if flags == libc::O_RDONLY {
+        if flags & libc::O_ACCMODE == libc::O_RDONLY {
             return Err(err!(libc::EBADF, "file handle is not open for writes"));
         }
 
+        let overwritable = fs.config.allow_overwrite && (flags & libc::O_TRUNC != 0);
         let handle = fs
             .superblock
-            .write(&fs.client, ino, lookup.inode.parent(), pid, fs.config.allow_overwrite)
+            .write(&fs.client, ino, lookup.inode.parent(), pid, overwritable)
```

Comment:
We should probably have different log messages for the "overwrite not enabled" and "overwrite enabled but not `O_TRUNC`" cases.